app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/service/DownloadIntentService.java

```diff
@@ -0,0 +1,123 @@
+package at_challenge.deaguiar.daniel.atchalleng.service;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.ResultReceiver;
+import android.util.Base64;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.DefaultHttpClient;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+/**
+ * DownloadIntentService
+ * Responsible to request data from server
+ *
+ * @author Daniel Besen de Aguiar
+ */
+public class DownloadIntentService extends IntentService {
+
+    public static final String INTENT_ID = "id";
+    public static final String URL_BUS_STOP = "url_stop";
+    public static final String URL_BUS_DEPARTURE = "url_departure";
+    public static final String RESULT_BUS = "result_bus";
+    public static final String RESULT_DEPARTURE = "result_departure";
+
+    public static final int STATUS_FINISHED = 0;
+    public static final int STATUS_ERROR    = 1;
+
+    private final static String USERNAME = "WKD4N7YMA1uiM8V";
+    private final static String PASSWORD = "DtdTtzMLQlA0hk2C1Yi5pLyVIlAQ68";
+
+    public DownloadIntentService() {
+        super("FetchDataFromService");
+    }
+
+    @Override
+    protected void onHandleIntent(Intent intent) {
+
+        final ResultReceiver resultReceiver = intent.getParcelableExtra("receiver");
+
+        // Get id from intent
+        int id = intent.getIntExtra(INTENT_ID, 0);
+
+        // Get URL's from intent
+        String urlStop = intent.getStringExtra(URL_BUS_STOP);
+        String urlDepart = intent.getStringExtra(URL_BUS_DEPARTURE);
+
+        Bundle bundle = new Bundle();
+        String jsonValue = "{\"params\": {\"routeId\": \"" + id + "\"}}";
+
+        try {
+            String resultBus = "";
+            String resultDepart = "";
+
+            // Get response for Bus
+            HttpResponse responseBus = getHttpResponse(jsonValue, urlStop);
```

Comment:
I decided to show the info to user only if I could fetch all the data. So I
downloaded all the data, check if data is ok and them show it to the user.

2015-06-17 13:21 GMT-03:00 André Luiz Cardoso <notifications@github.com>:

> In
> app/src/main/java/at_challenge/deaguiar/daniel/atchalleng/service/DownloadIntentService.java
> <https://github.com/daniboy000/at_challenge/pull/1#discussion_r32644909>:
>
> > +        // Get id from intent
> > +        int id = intent.getIntExtra(INTENT_ID, 0);
> > +
> > +        // Get URL's from intent
> > +        String urlStop = intent.getStringExtra(URL_BUS_STOP);
> > +        String urlDepart = intent.getStringExtra(URL_BUS_DEPARTURE);
> > +
> > +        Bundle bundle = new Bundle();
> > +        String jsonValue = "{\"params\": {\"routeId\": \"" + id + "\"}}";
> > +
> > +        try {
> > +            String resultBus = "";
> > +            String resultDepart = "";
> > +
> > +            // Get response for Bus
> > +            HttpResponse responseBus = getHttpResponse(jsonValue, urlStop);
>
> Why are you calling the bus and departures APIs in the same class?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/daniboy000/at_challenge/pull/1/files#r32644909>.
>



-- 
Daniel Besen de Aguiar

"Those who wish to succeed must ask the right preliminary questions"


      Aristotle